fix: warnings fail the test

**Description**

If warning occurs during running a test, the very_good test command exits with code 69

e.g. output from GitHub Action:

```
Run very_good test --coverage -x 'presubmit-only' --test-randomize-ordering-seed random
Running "flutter test" in /home/runner/work/org/repo...
Shuffling test order with --test-randomize-ordering-seed=2795804525
Optimizing tests (4.0s)

00:00 ...
00:01 ...
00:02 ...
00:03 ...
00:04 ...
00:05 ...
00:06 ...
00:07 .../opt/hostedtoolcache/.pub-cache/hosted/pub.dartlang.org/
cached_network_image-3.2.0/lib/src/image_provider/cached_network_image_provider.dart:109:29: 
Warning: Operand of null-aware operation '?.' has type 'PaintingBinding' which excludes null.

Error: Process completed with exit code 69.
```

The original warning in cached_network_image is already tracked here: https://github.com/Baseflow/flutter_cached_network_image/issues/732. Nonetheless I think the warning should not fail the step.


**Additional Context**

Affects:
- very_good_cli 0.7.0
- Dart 2.17
- Flutter 3.0.0
- running on macOS and Ubuntu

--- a/.github/workflows/very_good_cli.yaml
+++ b/.github/workflows/very_good_cli.yaml
@@ -40,7 +40,7 @@ jobs:
       - name: Run Tests
         run: |
           flutter pub global activate coverage 1.2.0
-          flutter pub run test -j 1 -x pull-request-only -x e2e --coverage=coverage --test-randomize-ordering-seed random && dart run coverage:format_coverage --lcov --in=coverage --out=coverage/lcov.info --packages=.packages --report-on=lib
+          flutter pub run test -j 1 -x pull-request-only -x e2e --coverage=coverage --test-randomize-ordering-seed random && dart pub global run coverage:format_coverage --lcov --in=coverage --out=coverage/lcov.info --packages=.packages --report-on=lib
 
       - name: Check Code Coverage
         uses: VeryGoodOpenSource/very_good_coverage@v1.2.0

--- a/.github/workflows/very_good_cli.yaml
+++ b/.github/workflows/very_good_cli.yaml
@@ -39,7 +39,7 @@ jobs:
 
       - name: Run Tests
         run: |
-          flutter pub global activate coverage
+          flutter pub global activate coverage 1.2.0
           flutter pub run test -j 1 -x pull-request-only -x e2e --coverage=coverage --test-randomize-ordering-seed random && dart run coverage:format_coverage --lcov --in=coverage --out=coverage/lcov.info --packages=.packages --report-on=lib
 
       - name: Check Code Coverage

--- a/lib/src/cli/flutter_cli.dart
+++ b/lib/src/cli/flutter_cli.dart
@@ -379,8 +379,8 @@ Future<int> _flutterTest({
       }
     },
     onError: (Object error, StackTrace stackTrace) {
-      subscription.cancel();
-      completer.completeError(error, stackTrace);
+      stderr('$clearLine$error');
+      stderr('$clearLine$stackTrace');
     },
   );
 

--- a/test/src/cli/flutter_cli_test.dart
+++ b/test/src/cli/flutter_cli_test.dart
@@ -312,7 +312,7 @@ void main() {
         );
       });
 
-      test('throws when process fails (with cleanup)', () async {
+      test('exits with code 69 when process fails (with cleanup)', () async {
         final directory = Directory.systemTemp.createTempSync();
         final testDirectory = Directory(p.join(directory.path, 'test'))
           ..createSync();
@@ -322,22 +322,20 @@ void main() {
             .writeAsStringSync(testContents);
         await expectLater(
           Flutter.test(cwd: directory.path, optimizePerformance: true),
-          throwsA(
-            '''Error: Couldn't resolve the package 'test' in 'package:test/test.dart'.''',
-          ),
+          completion(equals([69])),
         );
         await File(
           p.join(testDirectory.path, '.test_runner.dart'),
         ).ensureDeleted();
       });
 
-      test('throws when there is no test directory', () {
+      test('exits with code 69 when there is no test directory', () {
         final directory = Directory.systemTemp.createTempSync();
         File(p.join(directory.path, 'pubspec.yaml')).writeAsStringSync(pubspec);
 
         expectLater(
           Flutter.test(cwd: directory.path),
-          throwsA('Test directory "test" not found.'),
+          completion(equals([69])),
         );
       });
 

